main.js

```diff
@@ -1,4 +1,4 @@
-var scope = (typeof self !== "undefined" && self) || window;
+var scope = (typeof self !== "undefined" && self) || this;
```

Comment:
In fact, you could just do `var scope = this;`, which works in workers and in regular web pages.